[4.3] Workflow Transitions

brianteeman opened the pull request:
Recently there have been several people posting that they don't understand Workflows. I was reviewing my notes from the Workflows presentation I gave this year at JoomlaDay fr and I realised that we can easily remove one area of confusion.

### Summary of Changes
Add a title to the options so that it is clear what they are for. It was confusing before because the displayed value is a stage but the options are a transition. 

Bizaarely we must have realised this before because the Action button already adds some clarification by saying "Run Transitions"
![image](https://user-images.githubusercontent.com/1296369/204273117-67472721-8fba-41ca-b040-565253ddd219.png)

**NOTE: If you are using the workflow from the sample data the problems this PR fixes are more obvious**

### Testing Instructions
1. Enable workflows
2. in the article list view click on the stage
3. in the article edit view click on the workflow


### Actual result BEFORE applying this Pull Request
In the list view it is not clear that there are two different things in the list - current stage name and transitions
![image](https://user-images.githubusercontent.com/1296369/204276329-8ab33500-5841-4a11-be9d-d43a461786ea.png)


In the article view it says Workflow but it should say Workflow Stage and it is not clear that there are two different things in the list - current stage name and transitions
![image](https://user-images.githubusercontent.com/1296369/204276179-851a62c1-fba8-4b81-9ea4-b84f7cf71481.png)


### Expected result AFTER applying this Pull Request

Article List View
![image](https://user-images.githubusercontent.com/1296369/204275857-b4e9e5e7-575d-49dd-8e26-7e894900136a.png)



Article Edit View
![image](https://user-images.githubusercontent.com/1296369/204275951-4a5bf7d2-37d5-4577-a2ab-70929b7a71e6.png)


### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
screenshots will need updating

- [x] No documentation changes for manual.joomla.org needed

brianteeman committed:
[4.2] Workflow Transitions

Recently there have been several people posting that they don't understand Workflows. I was reviewing my notes from the Workflows presentation I gave this year at JoomlaDay fr and I realised that we can easily remove one area of confusion.

brianteeman committed:
stages

brianteeman committed:
Merge branch '4.3-dev' into transitions

viocassel commented:
I have tested this item :white_check_mark: successfully on cf92a8d226b1f65b11f77cd97c0f4f4f2b4565f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39325">issues.joomla.org/tracker/joomla-cms/39325</a>.</sub>

jwaisner commented:
I have tested this item :white_check_mark: successfully on cf92a8d226b1f65b11f77cd97c0f4f4f2b4565f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39325">issues.joomla.org/tracker/joomla-cms/39325</a>.</sub>

jwaisner commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39325">issues.joomla.org/tracker/joomla-cms/39325</a>.</sub>

obuisard commented:
I did not catch this one in time, but the fields could have been modified to extend GroupediistField rather than ListField, where grouping of options is handled more elegantly and would prevent the hard coding of <OPTGROUP>

brianteeman commented:
Do you want me to look at updating this pr with that?

brianteeman commented:
will do boss

brianteeman committed:
Merge branch '4.3-dev' into transitions

brianteeman commented:
Having looked at the code I am going to pass on making the change myself and stick with my pr. Feel free of course to reject it and to submit your own.

obuisard commented:
> Having looked at the code I am going to pass on making the change myself and stick with my pr. Feel free of course to reject it and to submit your own.

@brianteeman I do not want to close this PR because of the work that you put into it and the user benefits. 
However, I still think the TransitionField.php should be updated. 
There are 5 lines of code to modify and 2 to remove.
I can modify your file and commit the changes so we can save ourselves some time. Ok with you?

brianteeman commented:
I would prefer it if you did it as a pr to my branch

obuisard commented:
> I would prefer it if you did it as a pr to my branch

Ok, I can do that.

obuisard committed:
Merge branch '4.3-dev' into transitions

obuisard committed:
TransitionsField extends from Groupedlist rather than List (#379)

* Update TransitionField.php

Extend Groupedlist rather than List

* Update TransitionField.php

Removed spaces

* Update TransitionField.php

Test if items are present before creating a group

brianteeman committed:
update comment

brianteeman commented:
This will need you tests

obuisard commented:
I have tested this item :white_check_mark: successfully on 9cbbc25db62e042eb1b3478757ef5b546d159411<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39325">issues.joomla.org/tracker/joomla-cms/39325</a>.</sub>

obuisard commented:
Jacob @jwaisner, would you mind testing the PR once more as the Transition field has been updated to extend Groupedlist? Thank you very much!

obuisard commented:
Vio @viocassel, we need a new test for this updated PR, do you mind re-testing? Thank you!

viocassel commented:
I have tested this item :white_check_mark: successfully on 9cbbc25db62e042eb1b3478757ef5b546d159411<br /><br />👍<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39325">issues.joomla.org/tracker/joomla-cms/39325</a>.</sub>

Quy commented on administrator/language/en-GB/com_content.ini:
```diff
@@ -189,6 +190,7 @@ COM_CONTENT_URL_FIELD_BROWSERNAV_LABEL="URL Target Window"
 COM_CONTENT_URL_FIELD_C_BROWSERNAV_LABEL="URL C Target Window"
 COM_CONTENT_WARNING_PROVIDE_VALID_NAME="Please provide a valid, non-blank title."
 COM_CONTENT_WORKFLOW="Workflow"
+COM_CONTENT_WORKFLOW_STAGE="Workflow Stage"
 COM_CONTENT_WORKFLOW_NOT_FOUND="No default workflow available, please define one or contact an administrator."
```
```suggestion
COM_CONTENT_WORKFLOW_NOT_FOUND="No default workflow available, please define one or contact an administrator."
COM_CONTENT_WORKFLOW_STAGE="Workflow Stage"
```

obuisard replied:
Good catch!

brianteeman committed:
Update com_content.ini

obuisard committed:
Merge branch '4.3-dev' into transitions

obuisard commented:
Thank you Brian @brianteeman !

brianteeman commented:
thx

Quy commented:
@brianteeman Can you please look into this for the frontend? Thanks.

![39325-workflow](https://user-images.githubusercontent.com/368084/210890909-09665f78-dbc1-47d7-bb58-6f270f99cf44.jpg)
